great_expectations/experimental/datasources/config.py

```diff
@@ -2,32 +2,47 @@
 from __future__ import annotations
 
 import logging
+import pathlib
 from pprint import pformat as pf
-from typing import Dict, Type
+from typing import TYPE_CHECKING, Dict, List, Type, Union
 
-from pydantic import Extra, Field, validator
+from pydantic import Extra, Field, ValidationError, validator
 
 from great_expectations.experimental.datasources.experimental_base_model import (
     ExperimentalBaseModel,
 )
 from great_expectations.experimental.datasources.interfaces import Datasource
 from great_expectations.experimental.datasources.sources import _SourceFactories
 
+if TYPE_CHECKING:
+    from pydantic.error_wrappers import ErrorDict as PydanticErrorDict
+
+
 LOGGER = logging.getLogger(__name__)
 
 
 _ZEP_STYLE_DESCRIPTION = "ZEP Experimental Datasources"
 
+_MISSING_XDATASOURCES_ERRORS: PydanticErrorDict = {
+    "loc": ("xdatasources",),
+    "msg": "field required",
+    "type": "value_error.missing",
+}
+
 
 class GxConfig(ExperimentalBaseModel):
     """Represents the full new-style/experimental configuration file."""
 
-    datasources: Dict[str, Datasource] = Field(..., description=_ZEP_STYLE_DESCRIPTION)
+    xdatasources: Dict[str, Datasource] = Field(..., description=_ZEP_STYLE_DESCRIPTION)
+
+    @property
+    def datasources(self) -> Dict[str, Datasource]:
+        return self.xdatasources
```

Comment:
Adding this property required me to remove the custom properties serialization logic in `experimental_base_model.py`.

It looked like we weren't using it for anything so opted to remove it rather than add some additional `don't serialize this 'datasources' property` logic.

Reply:
👍. We were using it but I think the need got removed by adding a prevalidation check instead of a custom `__init__` where we some property related logic.